include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -107,22 +102,78 @@ inline void InitApi() noexcept { Global<void>::api_ = OrtGetApiBase()->GetApi(OR
 //   // ...
 // }
 //
-inline void InitApi(const OrtApi* api) noexcept { Global<void>::api_ = api; }
-#else
-#if defined(_MSC_VER) && !defined(__clang__)
-#pragma warning(push)
-// "Global initializer calls a non-constexpr function." Therefore you can't use ORT APIs in the other global initializers.
-// Please define ORT_API_MANUAL_INIT if it conerns you.
-#pragma warning(disable : 26426)
-#endif
-const OrtApi* Global<T>::api_ = OrtGetApiBase()->GetApi(ORT_API_VERSION);
-#if defined(_MSC_VER) && !defined(__clang__)
-#pragma warning(pop)
+inline void InitApi(const OrtApi* api) noexcept;
+
+namespace detail {
+// This is used internally by the C++ API. This class holds the global
+// variable that points to the OrtApi.
+struct Global {
+  static const OrtApi* GetApi() noexcept {
+    if (!api_) {
+      api_ = DefaultInit();
+    }
+
+    return api_;
+  }
+
+ private:
+  inline static const OrtApi* api_ = nullptr;
```

Comment:
>inline static const OrtApi* api_ = nullptr; [](http://example.com/codeflow?start=2&length=43)

Two questions.
1) Have you considered this to be a function local static?
2) Is this going to be a separate instance in each compilation unit?

Reply:
2. The existing code uses a static member as well. The `inline` keyword I'm adding here allows me to specify the initialization inside the class definition. It doesn't change other semantics. That is, this could be written as
   ```cpp
   class Global {
     static const  OrtApi* api_;
   }
   Global::api_ = nullptr;
   ```

Reply:
Thanks for the additional explanation.
I have thought about it, and I would favor making it a function local static because during concurrent calls to `GetApi()`
1) it would make it easier to reason about the `api_` ptr initialization
2) alleviate some static object concerns during DLL loading.